src/components/TextField.js

```diff
@@ -18,26 +19,45 @@ export type Props = {
   // other react-native textInput props
 };
 
-const TextField = ({
-  error,
-  inputRef,
-  label,
-  onBlur,
-  onChange,
-  value,
-  ...props
-}: Props) => (
-  <View>
-    <FormLabel>{label}</FormLabel>
-    <FormInput
-      ref={inputRef}
-      onBlur={onBlur}
-      onChangeText={onChange}
-      value={value && value.toString()}
-      {...props}
-    />
-    <FormValidationMessage>{error}</FormValidationMessage>
-  </View>
-);
+class TextField extends React.Component<Props> {
+  _inputRef: ?FormInput;
+
+  _setRef = (ref: FormInput) => {
+    const { inputRef } = this.props;
```

Comment:
Shouldn't this always be a function?

Reply:
In practiise, yes. but in react api it can be string also, `React.Ref = (ref: lalal) => void | string` . So if I want to annotate inputRef in props as React.Ref i have to do this check =(